libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp

```diff
@@ -253,7 +253,16 @@ AP_AdvancedFailsafe::check(uint32_t last_valid_rc_ms)
             _state = STATE_DATA_LINK_LOSS;
             if (_wp_comms_hold) {
                 _saved_wp = mission.get_current_nav_cmd().index;
-                mission.set_current_cmd(_wp_comms_hold);
+
+                //option to not go back to the loss comm mission item if we already passed it
+                if (option_is_set(Option::CONTINUE_IF_PASSED_LOSS_COMM_WP) && 
+                   (mission.state() == AP_Mission::MISSION_RUNNING) &&
+                   (mission.get_current_nav_cmd().index > _wp_comms_hold)){
```

Comment:
```suggestion
                   (mission.get_in_landing_sequence())){
```

... as discussed elsewhere, checking for a numeric difference in waypoint numbers doesn't make sense when there are jumps involved.  I believe you said you didn't want to do the set-current-cmd when you were already on the way home, so just check directly.